library/psa_crypto.c

```diff
@@ -1227,7 +1227,7 @@ static psa_status_t psa_copy_key_material( const psa_key_slot_t *source,
     buffer_size = PSA_KEY_EXPORT_MAX_SIZE( source->type,
                                            psa_get_key_bits( source ) );
     buffer = mbedtls_calloc( 1, buffer_size );
-    if( buffer == NULL )
+    if( buffer == NULL && buffer_size != 0 )
```

Comment:
`psa_internal_export_key()`, `psa_import_key()`, and `mbedtls_free()` are well defined when `buffer` is `NULL` and `buffer_size` is `0`, but how about `mbedtls_platform_zeroize()` which might just call libc's `memset()`?

Reply:
I've made the zeroization conditional on the buffer being of a non-zero length.